src/main/java/appeng/client/render/cablebus/P2PTunnelFrequencyBakedModel.java

```diff
@@ -0,0 +1,142 @@
+
+package appeng.client.render.cablebus;
+
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.block.model.ItemOverrideList;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.vertex.VertexFormat;
+import net.minecraft.util.EnumFacing;
+
+import appeng.api.parts.IPartBakedModel;
+import appeng.api.util.AEColor;
+import appeng.util.Platform;
+
+
+public class P2PTunnelFrequencyBakedModel implements IBakedModel, IPartBakedModel
+{
+	private final VertexFormat format;
+	private final TextureAtlasSprite texture;
+
+	private final static Cache<Long, List<BakedQuad>> modelCache = CacheBuilder.newBuilder().maximumSize( 100 ).build();
+
+	private static final int[][] QUAD_OFFSETS = new int[][] {
+			{ 4, 10, 2 },
+			{ 10, 10, 2 },
+			{ 4, 4, 2 },
+			{ 10, 4, 2 }
+	};
+
+	public P2PTunnelFrequencyBakedModel( final VertexFormat format, final TextureAtlasSprite texture )
+	{
+		this.format = format;
+		this.texture = texture;
+	}
+
+	@Override
+	public List<BakedQuad> getPartQuads( Long partFlags, long rand )
+	{
+		try
+		{
+			return modelCache.get( partFlags, () ->
+			{
+				short frequency = 0;
+				boolean active = false;
+				if( partFlags != null )
+				{
+					frequency = (short) ( partFlags.longValue() & 0xffffL );
+					active = ( partFlags.longValue() & 0x10000L ) != 0;
+				}
+				return getQuadsForFrequency( frequency, active );
+			} );
+		}
+		catch( ExecutionException e )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)